### PR TITLE
feat: support negative tokens

### LINKED ISSUE #370: Negative token
Beta's missing support for negative tokens:
```
margin: -$4
```

In the alpha, this resolves to:
```
margin: calc(var(--space-4) * -1)
```

### PR DESCRIPTION
**Pull request by jonathantneal**
This PR adds the ability to use negative tokens like `-$sp1`.

This resolves #370.

### CODE REVIEW

File: packages/core/src/createGetComputedCss.js [hunk 1]
```diff
@@ -110,7 +110,10 @@ const createGetComputedCss = (
 							// write the value as string, conditionally converted as a number into a px, or as a token resolved into a custom property
 							(typeof data === 'number' && !isPossiblyUnitless(name) && data
 								? data + 'px'
-								: String(data).replace(/\$[$-\w]+/g, (token) => 'var(-' + (!/.\$/.test(token) && name in themeMap ? '-' + themeMap[name] : '') + token.replace(/\$/g, '-') + ')')) +
+								: String(data).replace(
+										/(-)?(\$[$-\w]+)/g,
+										($0, negative, token) => (negative ? 'calc(' : '') + 'var(-' + (!/.[$]/.test(token) && name in themeMap ? '-' + themeMap[name] : '') + token.replace(/[$]/g, '-') + ')' + (negative ? '*-1)' : ''),
+								  )) +
```
Comment (peduarte) [inline packages/core/src/createGetComputedCss.js:116]: You sir are a regex genius

File: packages/core/tests/universal-tokens.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+						sp2: '20px',
+						sp3: '30px',
+					},
+				},
+			})
+
+			global({
+				article: {
+					marginLeft: '-$sizes$sp1',
+					width: '$sp1',
```
Comment (peduarte) [inline packages/core/tests/universal-tokens.js:95]: Is it worth adding a test for a simple negative token, without prefixing the scale?

**Review by peduarte — APPROVED** [review]

File: packages/core/tests/universal-tokens.js (hunk 2 above)
Comment (jonathantneal) [reply packages/core/tests/universal-tokens.js:115]: Yes it is. Thanks for the call out. Added in https://github.com/modulz/stitches/pull/411/commits/f4b27d37d4d83dd42e417a2a986b2fc682e492a8

**Comment by jonathantneal** [comment]
Thanks for the review. We’ll discuss this further tomorrow. Holding off on pushing this into today’s canary. :smile_cat:
